core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1941,11 +1941,19 @@ class KafkaApis(val requestChannel: RequestChannel,
     // with client authentication which is performed at an earlier stage of the connection where the
     // ApiVersionRequest is not available.
     val apiVersionRequest = request.body[ApiVersionsRequest]
+    val expectedClientSoftwareNames = config.expectedClientSoftwareNames
```

Comment:
Do we need any null checks here?

Reply:
If you check KafkaConfig.scala, there are default values defined, so unless you explicitly override it to null, this will not be null. But wouldn't harm to add null check.